neonvm/runner/main.go

```diff
@@ -532,17 +548,230 @@ func main() {
 		qemuCmd = append(qemuCmd, "-incoming", fmt.Sprintf("tcp:0:%d", vmv1.MigrationPort))
 	}
 
-	go terminateQemuOnSigterm(vmSpec.QMP)
-	if err := execFg(QEMU_BIN, qemuCmd...); err != nil {
-		log.Fatal(err)
+	// leading slash is important
+	cgroupPath := fmt.Sprintf("/%s-vm-runner", vmStatus.PodName)
+
+	if err := setCgroupLimit(qemuCPUs.use, cgroupPath); err != nil {
+		log.Fatalf("Failed to set cgroup limit: %s", err)
+	}
+	defer cleanupCgroup(cgroupPath)
+	ctx, cancel := context.WithCancel(context.Background())
+	wg := sync.WaitGroup{}
+
+	wg.Add(1)
+	go terminateQemuOnSigterm(ctx, vmSpec.QMP, &wg)
+	wg.Add(1)
+	go listenForCPUChanges(ctx, vmSpec.RunnerPort, cgroupPath, &wg)
+
+	args := append([]string{"-g", fmt.Sprintf("cpu:%s", cgroupPath), QEMU_BIN}, qemuCmd...)
+	log.Printf("build qemu args: %s", args)
```

Comment:
Not sure what "build qemu args" is referring to here. Seems like this is actually the arguments to cgexec? +1 to logging this, but description should be more clear, I think.